src/mon/HealthMonitor.cc

```diff
@@ -159,6 +160,10 @@ bool HealthMonitor::prepare_update(MonOpRequestRef op)
   dout(7) << "prepare_update " << *m
 	  << " from " << m->get_orig_source_inst() << dendl;
   switch (m->get_type()) {
+  case MSG_MON_HEALTH:
```

Comment:
you'll also need to handle the message in `Monitor::dispatch_op()`, to redirect it to the HealthMonitor.